bind/flag.go

```diff
@@ -279,17 +279,25 @@ func HTTPServerConfig(fs *pflag.FlagSet, cfg *forwarder.HTTPServerConfig, prefix
 	fs.VarP(anyflag.NewValueWithRedact[*url.Userinfo](cfg.BasicAuth, &cfg.BasicAuth, forwarder.ParseUserinfo, RedactUserinfo),
 		namePrefix+"basic-auth", "", "<username[:password]>"+
 			"Basic authentication credentials to protect the server. ")
+}
 
-	httpLogModes := []httplog.Mode{
-		httplog.None,
-		httplog.ShortURL,
-		httplog.URL,
-		httplog.Headers,
-		httplog.Body,
-		httplog.Errors,
+func HTTPLogConfig(fs *pflag.FlagSet, logcfg *httplog.Config) {
+	valid := make([]string, 0, len(logcfg.Modules()))
```

Comment:
I do not see why the config should know the modules why not pass the modules directly as a parameter here?

Reply:
I get it now, this way it can validate.